eureka2-ext/eureka2-eureka1x-rest-api/src/main/java/com/netflix/eureka2/eureka1x/rest/Eureka1xRedirectRequestHandler.java

```diff
@@ -0,0 +1,123 @@
+            .or()
+            .protocolType(ProtocolType.IPv4)
+            .secure(false)
+            .serviceLabel(com.netflix.eureka2.Names.EUREKA_HTTP);
+
+    private final ExtensionContext context;
+
+    private Subscription subscription;
+    private volatile List<InstanceInfo> readServers;
+    private volatile int serverIndex;
+
+    @Inject
+    public Eureka1xRedirectRequestHandler(ExtensionContext context,
+                                          EurekaHttpServer httpServer) {
+        this.context = context;
+        httpServer.connectHttpEndpoint(ROOT_PATH, this);
+    }
+
+    @PostConstruct
+    public void start() {
+        subscription = context.getLocalRegistry()
+                .forInterest(Interests.forVips(context.getConfig().getReadClusterVipAddress()))
+                .compose(ChangeNotifications.<InstanceInfo>delineatedBuffers())
+                .compose(ChangeNotifications.<InstanceInfo>snapshots())
+                .subscribe(new Subscriber<Set<InstanceInfo>>() {
+                    @Override
+                    public void onCompleted() {
+                        logger.info("Read server subscription completed");
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        logger.info("Read server stream terminated with an error", e);
+                    }
+
+                    @Override
+                    public void onNext(Set<InstanceInfo> instanceInfos) {
+                        logger.info("Updating read server cluster to: {}", instanceInfos);
+                        readServers = new ArrayList<InstanceInfo>(instanceInfos);
+                    }
+                });
+    }
+
+    @PreDestroy
+    public void shutdown() {
+        if (subscription != null) {
+            subscription.unsubscribe();
+        }
+    }
+
+    @Override
+    protected Observable<Void> dispatch(HttpServerRequest<ByteBuf> request, HttpServerResponse<ByteBuf> response) throws Exception {
+        List<InstanceInfo> currentServers = readServers;
+        if (currentServers != null && !currentServers.isEmpty()) {
+            InstanceInfo next = currentServers.get(serverIndex % currentServers.size());
+            serverIndex = (serverIndex + 1) % currentServers.size();
+
+            return redirectTo(next, request, response);
+        }
+        response.setStatus(HttpResponseStatus.SERVICE_UNAVAILABLE);
+        return Observable.empty();
+    }
+
+    private Observable<Void> redirectTo(InstanceInfo readServerInfo,
+                                        HttpServerRequest<ByteBuf> request,
+                                        HttpServerResponse<ByteBuf> response) {
+        ServiceEndpoint serviceEndpoint = HTTP_SERVICE_SELECTOR.returnServiceEndpoint(readServerInfo);
+
+        StringBuilder redirectBuilder = new StringBuilder("http://")
+                .append(serviceEndpoint.getAddress().getIpAddress())
```

Comment:
It is a bit tricky here, as we redirect clients which might be inside or outside.
Probably best default would be:
1. public hostname
2. if not available private ip
This would work best in AWS, but for generic cloud deployment might be too restrictive

Possibly we need some extensibility here, where the actual choice could be based on cloud type, and client IP address.